### PR TITLE
LPS-202152

### PR DESCRIPTION
**Pull request by icgsbr**
https://liferay.atlassian.net/browse/LPS-202152

### CODE REVIEW

**Comment by icgsbr** [comment]
ci:test:sf

**Comment by icgsbr** [comment]
ci:test:stable

**Comment by icgsbr** [comment]
ci:test:relevant

**Comment by icgsbr** [comment]
ci:test:forms

**Comment by icgsbr** [comment]
Hi @marcelabc !

Since the main tests passed and the failed ones are not related to the changes that were made, can we forward this to bchan?

**Comment by marcelabc** [comment]
Hi @icgsbr,
We should wait for the review. Sorry for the delay, I'm starting the review now :smile:

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/AddFormInstanceRecordMVCActionCommand.java
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-web/src/main/java/com/liferay/dynamic/data/mapping/form/web/internal/portlet/action/AddFormInstanceRecordMVCActionCommand.java:209]: Hey @icgsbr 
I was doing research on Forms, and I found this code:
```
DDMFormInstanceVersion latestDDMFormInstanceVersion =
	_ddmFormInstanceVersionLocalService.
	        getLatestFormInstanceVersion(
		        ddmFormInstance.getFormInstanceId(),WorkflowConstants.STATUS_APPROVED);

DDMStructureVersion ddmStructureVersion =
	latestDDMFormInstanceVersion.getStructureVersion();

ddmForm = ddmStructureVersion.getDDMForm();
```
I believe this code does what you are looking for (the latest published structure version). 
Could you try it? If it works, you won't have to add a new method.

**Review by marcelabc — CHANGES_REQUESTED** [review]
Hi 
I finished my review. Let me know if you have questions :smile:

**Comment by marcelabc** [comment]
Hey @icgsbr,
I saw that you push the commits, but you didn't remove the label. Do you want another review round?

**Comment by icgsbr** [comment]
> Hey @icgsbr, I saw that you push the commits, but you didn't remove the label. Do you want another review round?

Hi @marcelabc,

I'm still taking a look at it. From what I saw and tested, the method that you found didn't work as expected, meaning that the new method creation is indeed necessary. Yet, I'm still trying to make it work in order to exhaust my efforts.

**Comment by icgsbr** [comment]
ci:test:stable

**Comment by icgsbr** [comment]
ci:test:relevant

**Comment by icgsbr** [comment]
ci:test:forms

**Comment by icgsbr** [comment]
ci:reevaluate:1387401_9802

**Comment by icgsbr** [comment]
ci:test:stable

**Comment by icgsbr** [comment]
Hi @marcelabc !

From what I've seen, the failures are not related to the changes. Could you please confirm it and see if this pull is eligible to be sent to Brian?

**Comment by marcelabc** [comment]
ci:reevaluate:1387401_9801

**Comment by icgsbr** [comment]
ci:forward:force

**Pull request closed without merging**
